zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -436,15 +445,24 @@ angular.module('zeppelinWebApp')
         // not applying emacs key binding while the binding override Ctrl-v. default behavior of paste text on windows.
       }
 
-      var sqlModeTest = /^%(\w*\.)?\wql/;
+      $scope.setParagraphMode = function(session, pos) {
+          var paragraphText = session.getValue();
```

Comment:
I will reuse the text value. Don't know what is the performance implication of calling it again.